modules/cloud-run/main.tf

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  prefix = var.prefix == null ? "" : "${var.prefix}-"
+  service_account_email = (
+    var.service_account_create
+    ? (
+      length(google_service_account.service_account) > 0
+      ? google_service_account.service_account[0].email
+      : null
+    )
+    : var.service_account
+  )
+
+  annotations = merge(var.ingress_settings == null ? {} : { "run.googleapis.com/ingress" = var.ingress_settings },
+    var.vpc_connector_config == null
+    ? {}
+    : try(var.vpc_connector_config.ip_cidr_range, null) == null
+    ? { "run.googleapis.com/vpc-access-connector" = var.vpc_connector_config.name }
+    : { "run.googleapis.com/vpc-access-connector" = google_vpc_access_connector.connector.0.id }
+    ,
+    try(var.vpc_connector_config.egress_settings, null) == null
+    ? {}
+  : { "run.googleapis.com/vpc-access-egress" = var.vpc_connector_config.egress_settings })
+}
+
+resource "google_vpc_access_connector" "connector" {
+  count         = try(var.vpc_connector_config.ip_cidr_range, null) == null ? 0 : 1
+  project       = var.project_id
+  name          = var.vpc_connector_config.name
+  region        = var.region
+  ip_cidr_range = var.vpc_connector_config.ip_cidr_range
+  network       = var.vpc_connector_config.network
+}
+
+resource "google_cloud_run_service" "service" {
+  provider = google-beta
+  project  = var.project_id
+  location = var.region
+  name     = "${local.prefix}${var.name}"
+
+  template {
+    spec {
+      dynamic "containers" {
+        for_each = var.containers == null ? [] : var.containers
```

Comment:
Does this work if you have more than one container?

Reply:
https://cloud.google.com/run/docs/reference/rest/v1/RevisionSpec According to the API you can pass a list, although I am not sure if it works as in the UI they only allow you to specify one container. For compatibility with the API I set it as an array.

Reply:
Fixed